pinot-spi/src/main/java/org/apache/pinot/spi/config/table/IndexingConfig.java

```diff
@@ -64,6 +64,7 @@ public class IndexingConfig extends BaseJsonConfig {
   // TODO: Add a new configuration related to the segment generation
   private boolean _autoGeneratedInvertedIndex;
   private boolean _createInvertedIndexDuringSegmentGeneration;
+  private boolean _createBloomFilterDuringSegmentGeneration;
```

Comment:
Don't add this. We can always generate bloom filter during segment generation for now. Adding this can make future change harder because we might not want to have per-index flag

Reply:
Ack. This flag will be replaced with "createAllIndexesDuringSegmentGeneration".

Just to be clear, as discussed with @npawar , For the first part of this change, we'd be introducing a "createAllIndexesDuringSegmentGeneration" flag. If that is set as true, we'd create all indexes during segment creation.
If its set to false, we'll only maintain status quo, and create only the indexes that we already create. Are we aligned on this?

Reply:
My suggestion is to not add a flag in this PR and always create the bloom filter index + range index if they are configured. It does change the current behavior, but that is the desired behavior and we don't want to complicate the config in this PR. The only exception we have right now is inverted index, and we should think of a migration plan for it in a separate thread.

Reply:
So you're suggesting we change this default behavior, and call it out in release notes as backward-incompat?
That sounds good. If we're doing this, then might as well also (in separate PR) 
1. deprecate inv index field, tag the PR as release-notes
2. call out in next release it's deprecated
3. in future release remove it

Reply:
though this could have deep store usage metrics impact for users, which might be undesirable in some cases. wouldn't adding a flag just be safer?

Reply:
I wouldn't call it backward-incompatible though. When we add a new index type, we always create them during segment creation if configured (e.g. text, fst, json, h3, null-vector), and inverted index is the only exception here. We don't create index for range and bloom filter right now is because we missed this part when adding the feature.

In the long term, we want to add a separate config to configure which index types to create during segment creation, and by default we should generate all indexes (deprecate the `createInvertedIndexDuringSegmentGeneration`)

Reply:
Have enabled bloomFilter and range index creation by default. Inverted index however, still gets created only through the config flag.